libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.java

```diff
@@ -358,31 +359,52 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 	}
 
 	@Override
-	public void finish() {
-        initAnalyticsManager();
+	public void finish(UserAccount userAccount) {
+        initAnalyticsManager(userAccount);
         final UserAccountManager userAccountManager = SalesforceSDKManager.getInstance().getUserAccountManager();
         final List<UserAccount> authenticatedUsers = userAccountManager.getAuthenticatedUsers();
         final int numAuthenticatedUsers = authenticatedUsers == null ? 0 : authenticatedUsers.size();
-        final int userSwitchType;
-        if (numAuthenticatedUsers == 1) {
 
-            // We've already authenticated the first user, so there should be one.
-            userSwitchType = UserAccountManager.USER_SWITCH_TYPE_FIRST_LOGIN;
-        } else if (numAuthenticatedUsers > 1) {
-
-            // Otherwise we're logging in with an additional user.
-            userSwitchType = UserAccountManager.USER_SWITCH_TYPE_LOGIN;
-        } else {
+        /*
+         * Sends user switch intents only if this login flow is not a login triggered due
+         * to an incoming authentication request from an SP app or first user to login on IDP.
+         * If it is an incoming SP request, we should add the user account but NOT switch to
+         * the user or send user switch intents unless it's the first user being logged in.
+         */
+        boolean isFirstUserOrNotIDPFlow = !SalesforceSDKManager.getInstance().isIDPAppLoginFlowActive()
```

Comment:
Doesn't send these intents out if the login was in the IDP context, unless it's the first user.